spec/factories/users.rb

```diff
@@ -3,5 +3,6 @@
     sequence(:name) { |n| "User#{n}" }
     sequence(:email) { |n| "user#{n}@test.com" }
     sequence(:password) { |n| "123456#{n}" }
+    role
```

Comment:
Please not do like this do not call factory within factory.

Reply:
It seems that such code was in some other PR. Add the role while creating the user in the tests